### PR TITLE
Fix #59180 salt.module.solaris_shadow is broken on py3

### LINKED ISSUE #59180: [BUG] shadow.set_password "a bytes-like object is required, not 'str'."
**Description**
Looks like there is some str vs bytes issues with the shadow.set_password module, which is used by user.present.

**Setup**
I already tracked it down to the module, so I can trigger is without a state.

```
[root@illumos]# salt-call shadow.set_password sjorge $5$unAsNuUI$3VOtsbWOobzp9Bia//PzFXWC3pb09t6IiQszQqF9YO3

Passed invalid arguments: a bytes-like object is required, not 'str'.

Usage:

    Set the password for a named user. The password must be a properly defined
    hash, the password hash can be generated with this command:
    ``openssl passwd -1 <plaintext password>``

    CLI Example:

    .. code-block:: bash

        salt '*' shadow.set_password root $1$UYCIxa628.9qXjpQCjM4a..

```

**Steps to Reproduce the behavior**
Do a call to shadow.set_password, seems to work fine when using python 2 but broken on python 3.7

**Expected behavior**
Update to /etc/shadows to happen.

**Screenshots**
n/a

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```
Salt Version:
          Salt: 3002.2

Dependency Versions:
          cffi: 1.14.4
      cherrypy: unknown
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 2.11.2
       libgit2: 1.1.0
      M2Crypto: 0.36.0
          Mako: 1.1.3
       msgpack: 1.0.0
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: 2.20
      pycrypto: 3.9.9
  pycryptodome: 3.9.9
        pygit2: 1.4.0
        Python: 3.7.9 (default, Dec 12 2020, 18:03:07)
  python-gnupg: 0.4.6
        PyYAML: 5.3.1
         PyZMQ: 19.0.2
         smmap: 3.0.4
       timelib: 0.2.4
       Tornado: 4.5.3
           ZMQ: 4.3.2

System Versions:
          dist: pkgsrc 5.11
        locale: utf-8
       machine: i86pc
       release: 5.11
        system: SunOS
       version: release: trunk 5.11
```
</details>

**Additional context**

### PR DESCRIPTION
**Pull request by sjorge**
### What does this PR do?
Swap an open call with `rb` to one with `r`.

This fixes an error about byte encoding when using python 3, not super happy with the fix though, as it seems to `simple`.
But it works and I don't have much time to write a better fix.

### What issues does this PR fix or reference?
Fixes: #59180

### Previous Behavior
```Passed invalid arguments: a bytes-like object is required, not 'str'.```

### New Behavior
No error and both the shadow state and module now work again on illumos.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
- [ ] Docs => I don't this this requires a doc change

- [X] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated
No, not sure how to write a test particular change. It's not like the flow or logic changed.
Pointers welcome and I can add one.

### Commits signed with GPG?
No, well it was a one line change so I did it in the browser, so actually yes... but I don't have a personal GPG key... ¯\_(ツ)_/¯

### CODE REVIEW

**Comment by sjorge** [comment]
I've added a change log entry, if tests are needed I'd love to get some pointers on how to this this change?

**Comment by sjorge** [comment]
before:
```
[root@smpy3test ~]# salt-call state.apply saltenv=development
[ERROR   ] An exception occurred in this state: Traceback (most recent call last):
  File "/opt/salt/lib/python3.7/site-packages/salt/state.py", line 2154, in call
    *cdata["args"], **cdata["kwargs"]
  File "/opt/salt/lib/python3.7/site-packages/salt/loader.py", line 2106, in wrapper
    return f(*args, **kwargs)
  File "/opt/salt/lib/python3.7/site-packages/salt/states/user.py", line 660, in present
    __salt__["shadow.set_password"](name, password)
  File "/opt/salt/lib/python3.7/site-packages/salt/modules/solaris_shadow.py", line 296, in set_password
    comps = line.strip().split(":")
TypeError: a bytes-like object is required, not 'str'

local:
  Name: Europe/Brussels - Function: timezone.system - Result: Clean Started: - 19:59:50.137237 Duration: 1.915 ms
  Name: tzreload - Function: cmd.wait - Result: Clean Started: - 19:59:50.139655 Duration: 0.799 ms
  Name: C - Function: locale.present - Result: Clean Started: - 19:59:50.143649 Duration: 70.121 ms
  Name: en_US.UTF-8 - Function: locale.present - Result: Clean Started: - 19:59:50.214227 Duration: 57.465 ms
  Name: nl_BE.UTF-8 - Function: locale.present - Result: Clean Started: - 19:59:50.272099 Duration: 57.856 ms
  Name: C - Function: locale.system - Result: Clean Started: - 19:59:50.331598 Duration: 17.279 ms
  Name: /etc/profile.d/50_MANPATH.sh - Function: file.managed - Result: Clean Started: - 19:59:50.357668 Duration: 26.448 ms
  Name: /etc/profile.d/50_PATH.sh - Function: file.managed - Result: Clean Started: - 19:59:50.384228 Duration: 2.025 ms
  Name: /etc/profile.d - Function: file.directory - Result: Clean Started: - 19:59:50.387061 Duration: 1.715 ms
  Name: /etc/skel/.profile - Function: file.managed - Result: Clean Started: - 19:59:50.388903 Duration: 10.624 ms
  Name: /root/.profile - Function: file.managed - Result: Clean Started: - 19:59:50.399647 Duration: 7.543 ms
  Name: /opt/local/etc/zprofile - Function: file.managed - Result: Clean Started: - 19:59:50.407340 Duration: 9.551 ms
  Name: tmux - Function: pkg.installed - Result: Clean Started: - 19:59:52.033580 Duration: 89.743 ms
  Name: vim - Function: pkg.installed - Result: Clean Started: - 19:59:52.123678 Duration: 30.461 ms
  Name: wheel - Function: group.present - Result: Clean Started: - 19:59:52.156247 Duration: 1.525 ms
  Name: admin - Function: user.absent - Result: Clean Started: - 19:59:52.160132 Duration: 0.888 ms
  Name: zsh - Function: pkg.installed - Result: Clean Started: - 19:59:52.162483 Duration: 31.46 ms
----------
          ID: base_user_sjorge
    Function: user.present
        Name: sjorge
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/opt/salt/lib/python3.7/site-packages/salt/state.py", line 2154, in call
                  *cdata["args"], **cdata["kwargs"]
                File "/opt/salt/lib/python3.7/site-packages/salt/loader.py", line 2106, in wrapper
                  return f(*args, **kwargs)
                File "/opt/salt/lib/python3.7/site-packages/salt/states/user.py", line 660, in present
                  __salt__["shadow.set_password"](name, password)
                File "/opt/salt/lib/python3.7/site-packages/salt/modules/solaris_shadow.py", line 296, in set_password
                  comps = line.strip().split(":")
              TypeError: a bytes-like object is required, not 'str'
     Started: 19:59:52.194551
    Duration: 23.826 ms
     Changes:
----------
          ID: base_user_sjorge
    Function: rbac.managed
        Name: sjorge
      Result: False
     Comment: One or more requisite failed: base.user.base_user_sjorge
     Started: 19:59:52.220187
    Duration: 0.005 ms
     Changes:
----------
          ID: base_user_sjorge
    Function: ssh_auth.manage
        Name: sjorge
      Result: False
     Comment: One or more requisite failed: base.user.base_user_sjorge
     Started: 19:59:52.221375
    Duration: 0.004 ms
     Changes:
  Name: sudo - Function: pkg.installed - Result: Clean Started: - 19:59:52.221461 Duration: 30.887 ms
  Name: /opt/local/etc/sudoers - Function: file.managed - Result: Clean Started: - 19:59:52.252475 Duration: 2.255 ms
  Name: /opt/local/etc/sudoers.d/wheel - Function: file.managed - Result: Clean Started: - 19:59:52.256009 Duration: 2.811 ms
  Name: /opt/local/etc/sudoers.d - Function: file.directory - Result: Clean Started: - 19:59:52.259316 Duration: 1.761 ms
  Name: /usr/sbin/route -p add -inet 0.0.0.0/0 10.23.30.1 -ifp net0 -static - Function: cmd.run - Result: Clean Started: - 19:59:52.261228 Duration: 686.833 ms
  Name: /usr/sbin/route -p add -inet6 ::/0 2a02:578:470f:30::1 -ifp net0 -static - Function: cmd.run - Result: Clean Started: - 19:59:52.948538 Duration: 35.038 ms
  Name: /etc/resolv.conf - Function: file.managed - Result: Clean Started: - 19:59:52.984021 Duration: 74.087 ms
  Name: /etc/hosts - Function: file.managed - Result: Clean Started: - 19:59:53.058273 Duration: 94.343 ms
  Name: /etc/motd - Function: file.managed - Result: Clean Started: - 19:59:53.152763 Duration: 2.216 ms
  Name: /opt/local/etc/salt/autosign_grains - Function: file.directory - Result: Clean Started: - 19:59:53.155117 Duration: 1.236 ms
  Name: python37 - Function: pkg.installed - Result: Clean Started: - 19:59:53.156513 Duration: 33.346 ms
  Name: py37-pip - Function: pkg.installed - Result: Clean Started: - 19:59:53.189984 Duration: 31.394 ms
  Name: py37-virtualenv - Function: pkg.installed - Result: Clean Started: - 19:59:53.221497 Duration: 29.987 ms
  Name: /opt/salt - Function: file.directory - Result: Clean Started: - 19:59:53.251609 Duration: 1.054 ms
  Name: build-essential - Function: pkg.installed - Result: Clean Started: - 19:59:53.252782 Duration: 31.765 ms
  Name: py37-cffi - Function: pkg.installed - Result: Clean Started: - 19:59:53.284672 Duration: 31.783 ms
  Name: py37-zmq - Function: pkg.installed - Result: Clean Started: - 19:59:53.316607 Duration: 30.115 ms
  Name: py37-pygit2 - Function: pkg.installed - Result: Clean Started: - 19:59:53.346841 Duration: 30.578 ms
  Name: py37-m2crypto - Function: pkg.installed - Result: Clean Started: - 19:59:53.377545 Duration: 31.432 ms
  Name: py37-crypto - Function: pkg.installed - Result: Clean Started: - 19:59:53.409123 Duration: 30.863 ms
  Name: py37-mako - Function: pkg.installed - Result: Clean Started: - 19:59:53.440119 Duration: 30.058 ms
  Name: /opt/salt - Function: virtualenv.managed - Result: Clean Started: - 19:59:53.471320 Duration: 3881.065 ms
  Name: /var/svc/manifest/pkgsrc/salt.xml - Function: file.managed - Result: Clean Started: - 19:59:57.353859 Duration: 7.133 ms
  Name: svccfg import /var/svc/manifest/pkgsrc/salt.xml - Function: cmd.watch - Result: Clean Started: - 19:59:57.362504 Duration: 1.112 ms
  Name: /opt/local/etc/salt/minion.d/_schedule.conf - Function: file.managed - Result: Clean Started: - 19:59:57.375617 Duration: 1.763 ms
  Name: /opt/local/etc/salt/minion.d/20_general.conf - Function: file.managed - Result: Clean Started: - 19:59:57.377485 Duration: 24.941 ms
  Name: /opt/local/etc/salt/minion.d/30_logging.conf - Function: file.managed - Result: Clean Started: - 19:59:57.402548 Duration: 10.793 ms
  Name: /opt/local/etc/salt/minion.d/10_master.conf - Function: file.managed - Result: Clean Started: - 19:59:57.413479 Duration: 11.391 ms
  Name: /opt/local/etc/salt/minion.d/11_masterless.conf - Function: file.managed - Result: Clean Started: - 19:59:57.424978 Duration: 12.434 ms
  Name: /opt/local/etc/salt/minion.d/40_network.conf - Function: file.managed - Result: Clean Started: - 19:59:57.437520 Duration: 9.347 ms
  Name: /opt/local/etc/salt/minion.d/40_startup.conf - Function: file.managed - Result: Clean Started: - 19:59:57.446980 Duration: 9.516 ms
  Name: /opt/local/etc/salt/minion.d - Function: file.directory - Result: Clean Started: - 19:59:57.459281 Duration: 2.149 ms
  Name: salt:minion - Function: service.running - Result: Clean Started: - 19:59:57.465116 Duration: 274.296 ms
  Name: /opt/local/etc/salt/minion - Function: file.absent - Result: Clean Started: - 19:59:57.739935 Duration: 1.308 ms
  Name: zones/data/smtest/gitea - Function: zfs.filesystem_present - Result: Clean Started: - 19:59:57.741459 Duration: 89.423 ms
  Name: gitea - Function: pkg.installed - Result: Clean Started: - 19:59:57.832043 Duration: 34.652 ms
  Name: /var/db/gitea - Function: file.directory - Result: Clean Started: - 19:59:57.868178 Duration: 5.439 ms
  Name: /var/db/gitea/indexers - Function: file.directory - Result: Clean Started: - 19:59:57.874273 Duration: 1.343 ms
  Name: /var/db/gitea/repositories - Function: file.directory - Result: Clean Started: - 19:59:57.876251 Duration: 1.308 ms
  Name: /var/db/gitea/data/avatars - Function: file.directory - Result: Clean Started: - 19:59:57.878250 Duration: 1.506 ms
  Name: /var/db/gitea/data/lfs - Function: file.directory - Result: Clean Started: - 19:59:57.880374 Duration: 1.371 ms
  Name: /var/db/gitea/data/sessions - Function: file.directory - Result: Clean Started: - 19:59:57.882376 Duration: 1.318 ms
  Name: /var/db/gitea/ssl - Function: file.directory - Result: Clean Started: - 19:59:57.884315 Duration: 1.519 ms
  Name: /opt/local/etc/gitea/conf/app.ini - Function: ini.options_present - Result: Clean Started: - 19:59:57.886823 Duration: 41.63 ms
  Name: gitea - Function: service.running - Result: Clean Started: - 19:59:57.929280 Duration: 241.536 ms
  Name: /opt/local/etc/salt/master - Function: file.absent - Result: Clean Started: - 19:59:58.180838 Duration: 1.274 ms
  Name: /opt/local/etc/salt/master.d/10_fileserver.conf - Function: file.managed - Result: Clean Started: - 19:59:58.187394 Duration: 18.076 ms
  Name: /opt/local/etc/salt/master.d/11_filserver_limits.conf - Function: file.managed - Result: Clean Started: - 19:59:58.205595 Duration: 10.88 ms
  Name: /opt/local/etc/salt/master.d/50_general.conf - Function: file.managed - Result: Clean Started: - 19:59:58.216607 Duration: 10.499 ms
  Name: /opt/local/etc/salt/master.d/30_logging.conf - Function: file.managed - Result: Clean Started: - 19:59:58.227220 Duration: 10.086 ms
  Name: /opt/local/etc/salt/master.d/40_network.conf - Function: file.managed - Result: Clean Started: - 19:59:58.237420 Duration: 9.952 ms
  Name: /opt/local/etc/salt/master.d/50_nodegroups.conf - Function: file.managed - Result: Clean Started: - 19:59:58.247488 Duration: 10.3 ms
  Name: /opt/local/etc/salt/master.d/20_security.conf - Function: file.managed - Result: Clean Started: - 19:59:58.257903 Duration: 10.136 ms
  Name: /opt/local/etc/salt/master.d - Function: file.directory - Result: Clean Started: - 19:59:58.270978 Duration: 2.273 ms
  Name: salt:master - Function: service.running - Result: Clean Started: - 19:59:58.276997 Duration: 259.545 ms

Summary for local
-------------
Succeeded: 72
Failed:     3
-------------
Total states run:     75
Total run time:    6.696 s
```

after:
```
[root@smpy3test ~]# salt-call state.apply saltenv=development
local:
  Name: Europe/Brussels - Function: timezone.system - Result: Clean Started: - 19:58:13.435242 Duration: 2.116 ms
  Name: tzreload - Function: cmd.wait - Result: Clean Started: - 19:58:13.437988 Duration: 1.201 ms
  Name: C - Function: locale.present - Result: Clean Started: - 19:58:13.442467 Duration: 70.467 ms
  Name: en_US.UTF-8 - Function: locale.present - Result: Clean Started: - 19:58:13.513372 Duration: 58.975 ms
  Name: nl_BE.UTF-8 - Function: locale.present - Result: Clean Started: - 19:58:13.572837 Duration: 58.849 ms
  Name: C - Function: locale.system - Result: Clean Started: - 19:58:13.633576 Duration: 18.012 ms
  Name: /etc/profile.d/50_MANPATH.sh - Function: file.managed - Result: Clean Started: - 19:58:13.662087 Duration: 28.4 ms
  Name: /etc/profile.d/50_PATH.sh - Function: file.managed - Result: Clean Started: - 19:58:13.690635 Duration: 2.174 ms
  Name: /etc/profile.d - Function: file.directory - Result: Clean Started: - 19:58:13.693639 Duration: 1.961 ms
  Name: /etc/skel/.profile - Function: file.managed - Result: Clean Started: - 19:58:13.695734 Duration: 10.518 ms
  Name: /root/.profile - Function: file.managed - Result: Clean Started: - 19:58:13.706374 Duration: 6.987 ms
  Name: /opt/local/etc/zprofile - Function: file.managed - Result: Clean Started: - 19:58:13.713482 Duration: 9.037 ms
  Name: tmux - Function: pkg.installed - Result: Clean Started: - 19:58:15.327514 Duration: 84.683 ms
  Name: vim - Function: pkg.installed - Result: Clean Started: - 19:58:15.412568 Duration: 32.383 ms
  Name: wheel - Function: group.present - Result: Clean Started: - 19:58:15.446957 Duration: 1.937 ms
  Name: admin - Function: user.absent - Result: Clean Started: - 19:58:15.451322 Duration: 0.738 ms
  Name: zsh - Function: pkg.installed - Result: Clean Started: - 19:58:15.453429 Duration: 32.799 ms
----------
          ID: base_user_sjorge
    Function: user.present
        Name: sjorge
      Result: True
     Comment: Updated user sjorge
     Started: 19:58:15.486761
    Duration: 36.212 ms
     Changes:
              ----------
              passwd:
                  XXX-REDACTED-XXX
  Name: sjorge - Function: rbac.managed - Result: Clean Started: - 19:58:15.524440 Duration: 2.449 ms
  Name: sjorge - Function: ssh_auth.manage - Result: Clean Started: - 19:58:15.528184 Duration: 4.653 ms
  Name: sudo - Function: pkg.installed - Result: Clean Started: - 19:58:15.532974 Duration: 31.87 ms
  Name: /opt/local/etc/sudoers - Function: file.managed - Result: Clean Started: - 19:58:15.564977 Duration: 2.219 ms
  Name: /opt/local/etc/sudoers.d/wheel - Function: file.managed - Result: Clean Started: - 19:58:15.568312 Duration: 2.796 ms
  Name: /opt/local/etc/sudoers.d - Function: file.directory - Result: Clean Started: - 19:58:15.571646 Duration: 1.812 ms
  Name: /usr/sbin/route -p add -inet 0.0.0.0/0 10.23.30.1 -ifp net0 -static - Function: cmd.run - Result: Clean Started: - 19:58:15.573617 Duration: 665.292 ms
  Name: /usr/sbin/route -p add -inet6 ::/0 2a02:578:470f:30::1 -ifp net0 -static - Function: cmd.run - Result: Clean Started: - 19:58:16.239454 Duration: 35.049 ms
  Name: /etc/resolv.conf - Function: file.managed - Result: Clean Started: - 19:58:16.275017 Duration: 73.235 ms
  Name: /etc/hosts - Function: file.managed - Result: Clean Started: - 19:58:16.348429 Duration: 95.122 ms
  Name: /etc/motd - Function: file.managed - Result: Clean Started: - 19:58:16.443705 Duration: 2.301 ms
  Name: /opt/local/etc/salt/autosign_grains - Function: file.directory - Result: Clean Started: - 19:58:16.446144 Duration: 1.229 ms
  Name: python37 - Function: pkg.installed - Result: Clean Started: - 19:58:16.447503 Duration: 33.775 ms
  Name: py37-pip - Function: pkg.installed - Result: Clean Started: - 19:58:16.481406 Duration: 30.353 ms
  Name: py37-virtualenv - Function: pkg.installed - Result: Clean Started: - 19:58:16.511915 Duration: 39.362 ms
  Name: /opt/salt - Function: file.directory - Result: Clean Started: - 19:58:16.551414 Duration: 1.102 ms
  Name: build-essential - Function: pkg.installed - Result: Clean Started: - 19:58:16.552645 Duration: 33.259 ms
  Name: py37-cffi - Function: pkg.installed - Result: Clean Started: - 19:58:16.586033 Duration: 30.724 ms
  Name: py37-zmq - Function: pkg.installed - Result: Clean Started: - 19:58:16.616885 Duration: 30.432 ms
  Name: py37-pygit2 - Function: pkg.installed - Result: Clean Started: - 19:58:16.647441 Duration: 32.912 ms
  Name: py37-m2crypto - Function: pkg.installed - Result: Clean Started: - 19:58:16.680483 Duration: 30.767 ms
  Name: py37-crypto - Function: pkg.installed - Result: Clean Started: - 19:58:16.711369 Duration: 30.221 ms
  Name: py37-mako - Function: pkg.installed - Result: Clean Started: - 19:58:16.741710 Duration: 30.875 ms
  Name: /opt/salt - Function: virtualenv.managed - Result: Clean Started: - 19:58:16.773989 Duration: 3904.892 ms
  Name: /var/svc/manifest/pkgsrc/salt.xml - Function: file.managed - Result: Clean Started: - 19:58:20.680280 Duration: 7.934 ms
  Name: svccfg import /var/svc/manifest/pkgsrc/salt.xml - Function: cmd.watch - Result: Clean Started: - 19:58:20.689885 Duration: 0.932 ms
  Name: /opt/local/etc/salt/minion.d/_schedule.conf - Function: file.managed - Result: Clean Started: - 19:58:20.703621 Duration: 1.906 ms
  Name: /opt/local/etc/salt/minion.d/20_general.conf - Function: file.managed - Result: Clean Started: - 19:58:20.705639 Duration: 24.608 ms
  Name: /opt/local/etc/salt/minion.d/30_logging.conf - Function: file.managed - Result: Clean Started: - 19:58:20.730387 Duration: 10.268 ms
  Name: /opt/local/etc/salt/minion.d/10_master.conf - Function: file.managed - Result: Clean Started: - 19:58:20.740772 Duration: 12.105 ms
  Name: /opt/local/etc/salt/minion.d/11_masterless.conf - Function: file.managed - Result: Clean Started: - 19:58:20.752997 Duration: 13.241 ms
  Name: /opt/local/etc/salt/minion.d/40_network.conf - Function: file.managed - Result: Clean Started: - 19:58:20.766372 Duration: 9.932 ms
  Name: /opt/local/etc/salt/minion.d/40_startup.conf - Function: file.managed - Result: Clean Started: - 19:58:20.776438 Duration: 11.099 ms
  Name: /opt/local/etc/salt/minion.d - Function: file.directory - Result: Clean Started: - 19:58:20.790385 Duration: 2.319 ms
  Name: salt:minion - Function: service.running - Result: Clean Started: - 19:58:20.795981 Duration: 248.15 ms
  Name: /opt/local/etc/salt/minion - Function: file.absent - Result: Clean Started: - 19:58:21.044647 Duration: 1.281 ms
  Name: zones/data/smtest/gitea - Function: zfs.filesystem_present - Result: Clean Started: - 19:58:21.046141 Duration: 82.864 ms
  Name: gitea - Function: pkg.installed - Result: Clean Started: - 19:58:21.129389 Duration: 35.356 ms
  Name: /var/db/gitea - Function: file.directory - Result: Clean Started: - 19:58:21.166170 Duration: 2.825 ms
  Name: /var/db/gitea/indexers - Function: file.directory - Result: Clean Started: - 19:58:21.169647 Duration: 1.348 ms
  Name: /var/db/gitea/repositories - Function: file.directory - Result: Clean Started: - 19:58:21.171620 Duration: 1.4 ms
  Name: /var/db/gitea/data/avatars - Function: file.directory - Result: Clean Started: - 19:58:21.173751 Duration: 1.317 ms
  Name: /var/db/gitea/data/lfs - Function: file.directory - Result: Clean Started: - 19:58:21.175748 Duration: 1.487 ms
  Name: /var/db/gitea/data/sessions - Function: file.directory - Result: Clean Started: - 19:58:21.177974 Duration: 1.522 ms
  Name: /var/db/gitea/ssl - Function: file.directory - Result: Clean Started: - 19:58:21.180220 Duration: 1.514 ms
  Name: /opt/local/etc/gitea/conf/app.ini - Function: ini.options_present - Result: Clean Started: - 19:58:21.182741 Duration: 39.284 ms
  Name: gitea - Function: service.running - Result: Clean Started: - 19:58:21.222873 Duration: 225.106 ms
  Name: /opt/local/etc/salt/master - Function: file.absent - Result: Clean Started: - 19:58:21.458607 Duration: 1.372 ms
  Name: /opt/local/etc/salt/master.d/10_fileserver.conf - Function: file.managed - Result: Clean Started: - 19:58:21.465044 Duration: 18.687 ms
  Name: /opt/local/etc/salt/master.d/11_filserver_limits.conf - Function: file.managed - Result: Clean Started: - 19:58:21.483864 Duration: 10.821 ms
  Name: /opt/local/etc/salt/master.d/50_general.conf - Function: file.managed - Result: Clean Started: - 19:58:21.494800 Duration: 10.022 ms
  Name: /opt/local/etc/salt/master.d/30_logging.conf - Function: file.managed - Result: Clean Started: - 19:58:21.504937 Duration: 10.495 ms
  Name: /opt/local/etc/salt/master.d/40_network.conf - Function: file.managed - Result: Clean Started: - 19:58:21.515547 Duration: 10.521 ms
  Name: /opt/local/etc/salt/master.d/50_nodegroups.conf - Function: file.managed - Result: Clean Started: - 19:58:21.526183 Duration: 10.237 ms
  Name: /opt/local/etc/salt/master.d/20_security.conf - Function: file.managed - Result: Clean Started: - 19:58:21.536553 Duration: 10.202 ms
  Name: /opt/local/etc/salt/master.d - Function: file.directory - Result: Clean Started: - 19:58:21.549609 Duration: 2.582 ms
  Name: salt:master - Function: service.running - Result: Clean Started: - 19:58:21.556463 Duration: 265.561 ms

Summary for local
-------------
Succeeded: 75 (changed=1)
Failed:     0
-------------
Total states run:     75
Total run time:    6.692 s
```

**Comment by sagetherage** [comment]
@dmurphy18 can you give a review and perhaps help with a test case?

**Comment by sjorge** [comment]
Bump, without this user management is mostly broken on solarisish systems.

**Comment by sagetherage** [comment]
Getting the latest from `master` and @waynew perhaps this is a good PR for the Test Clinic this week?

**Comment by dmurphy18** [comment]
@sjorge Was wondering what system you were running this on ?, a version-reports would be helpful

**Review by dmurphy18 — DISMISSED** [review]

**Comment by waynew** [comment]
@sagetherage / @sjorge I went ahead and added tests for this (also took the liberty of rebasing on master). Feel free to check out the Test Clinic recordings for the approach that I took :+1:

**Comment by dmurphy18** [comment]
@waynew Some of those tests appear to need a 'Not on Solaris' or vice-versa ?

**Comment by waynew** [comment]
@dmurphy18 Technically they could be skipped on other platforms, but the tests here aren't actually hitting `/etc/shadow` at all. It's probably not /necessary/ that they run on other platforms, since the underlying module only runs on solaris, but it's not necessarily a bad thing.

I went ahead and added some skip code that will skip if the underlying modules are missing - it should pass on the platforms that were missing pwd/spwd now :+1:

**Comment by sjorge** [comment]
I don’t think there are any automated test runs on Solarisish like platforms at all right? No Oracle Solaris or OmniOS (illumos)

**Comment by Ch3LL** [comment]
Correct, we do not have automated tests on those platforms. @waynew is there a way to ensure it is running on our automated tests?

**Comment by waynew** [comment]
One could manually check the output on jenkins :+1:

**Comment by Ch3LL** [comment]
Let me rephrase, I did check manually the tests and I do not see an instance where all the tests are running, because of them getting skipped, although I could be mistaken. Can we update the test or add a binary to ensure they are running?

**Comment by waynew** [comment]
@Ch3LL Wow, I must have just committed those changes without bothering to even test them. Doh!

I had a typo and I was just skipping everything since the skip flags were always set to True :facepalm: 

*now* it should actually be running all the tests on most of the Linux-y platforms. There may be some platforms like catalina and freebsd that are missing one or the other submodules so there's no module being imported to mock out, but... yeah. We should have at least one platform that is running all of these tests.
